scene/gui/color_picker.cpp

```diff
@@ -353,27 +654,77 @@ void ColorPicker::_value_changed(double) {
 		return;
 	}
 
-	if (hsv_mode_enabled) {
-		h = scroll[0]->get_value() / 360.0;
-		s = scroll[1]->get_value() / 100.0;
-		v = scroll[2]->get_value() / 100.0;
-		if (picker_type == SHAPE_OKHSL_CIRCLE) {
-			color.set_ok_hsl(h, s, v, Math::round(scroll[3]->get_value() / 255.0));
-		} else {
-			color.set_hsv(h, s, v, Math::round(scroll[3]->get_value() / 255.0));
-		}
+	Vector<float> components;
+	for (int i = 0; i < current_sliders_count; i++) {
+		components.push_back(scrolls[i]->get_value());
+	}
+	components.push_back(alpha_scroll->get_value());
+
+	color = selected_mode->get_color(components);
 
+	if (mode_option_button->get_selected() == HSV_MODE_INDEX || mode_option_button->get_selected() == OKHSL_MODE_INDEX) {
```

Comment:
Is this condition necessary?
You can set h/s/v even for modes that don't need it.

Reply:
Did this as it was checking for HSV mode previously but the call to `_set_pick_color` will do this anyway. Will remove this condition.

Reply:
Nope, removing the condition is causing this bug in RGB mode with OKHSL picker 
<img width="166" alt="image" src="https://user-images.githubusercontent.com/79760854/174430055-d1bed615-baeb-4676-bfcc-b11650404ae0.png">